--- a/graph_copy_with_names.icl
+++ b/graph_copy_with_names.icl
@@ -393,6 +393,7 @@ get_thunk_n_non_pointers d
 		# b_size = arity>>8;
 		= b_size;
 
+make_symbol_name :: !String !String !Int -> String;
 make_symbol_name "_system" descriptor_name prefix
 	= case descriptor_name of {
 		"INT"
@@ -617,6 +618,7 @@ replace_descs_by_desc_numbers_and_build_desc_tree i s n_descs array_desc desc_tr
 		#! d = get_D_from_string s (i+IF_INT_64_OR_32 16 8);
 		| d==0
 			= replace_descs_by_desc_numbers_and_build_desc_tree (i+(IF_INT_64_OR_32 24 12)) s n_descs array_desc desc_tree;
+		# d = d+array_desc;
 		# (s,n_descs,desc_tree) = store_desc_n_and_add_desc d (i+IF_INT_64_OR_32 16 8) s n_descs desc_tree;
 		#! l = get_D_from_string s (i+IF_INT_64_OR_32 8 4);
         | is_Int_D d
